### PR TITLE
Topics: add Simplicity

### PR DESCRIPTION
**Pull request by michaelfolkson**
At the very least I need to add hashtags (I couldn't find the hashtags to link to specific sections of previous newsletters). There may be other stuff I've missed too.

However, I think we need a Simplicity topics page as it has cropped up in the newsletter a number of times now and so hopefully this is at least a start.

### CODE REVIEW

**Comment by harding** [comment]
Concept ACK. Thanks!

I'd prefer a more detailed excerpt/detailed_summary, perhaps including information like:

- Planned for Elements
- Not proposed for Bitcoin directly in the near future (e.g. the next 24 months)
- Easier proofs of correctness than Script
- Maybe better resource accounting / better management of DoS risks
- Native merklized scripting
- Risks, e.g. more Ethereum-like losses due to poorly written contracts
- Costs, e.g. non-jetted scripts might be more expensive than current scripts
- More details about what jets are and why they're useful

In short, text that convinces me that this is worth considering (but without hiding any likely downsides).

@michaelfolkson Let me know if the anchor linking instructions in https://github.com/harding/bitcoinops.github.io/pull/15#issuecomment-668042421 are insufficient to address the links on this PR as well.

**Comment by bitschmidty** [comment]
Concept ACK as well! I could have used this link last week as well

**Comment by jnewbery** [comment]
Thanks for this @michaelfolkson . Will review once you've addressed @harding's review comments.

**Comment by jnewbery** [comment]
@michaelfolkson - are you planning to update this PR?

**Comment by michaelfolkson** [comment]
Yes I will work on this after I've finished this month's transcripts section. Thanks for the reminder and apologies for delay.

**Comment by michaelfolkson** [comment]
Pushed an additional commit that addresses most if not all of @harding suggestions and adds the correct links to prior newsletters.

File: _topics/en/simplicity.md [hunk 1]
```diff
@@ -10,14 +10,31 @@ categories:
 ## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
 excerpt: >
   **Simplicity** is a work in progress low level programming language designed
-  to offer greater functionality than Bitcoin Script. It will require a future
-  soft fork to be enabled on Bitcoin.
+  to offer greater flexibility and expressiveness than Bitcoin Script whilst
+  allowing you to verify the safety, security and costs of your program. It
+  offers native Merklized scripting, formal semantics and type checking.
+  It is not Turing complete but it can verify the execution of Turing complete
+  programs. Although Simplicity provides certain proofs of correctness,
```
Comment (harding) [inline _topics/en/simplicity.md:17]: My guess is that you're trying to say that solutions generated by a turing complete system don't necessary need to be evaluated by a turing complete system in order to be proven correct.  For that, you verify *results* not *execution*.  Think of verifying results as your boss inspecting something you built and verifying execution as your boss watching you while you work.  The former can be effectively done with a static checklist; the later can only be effectively done by someone who can understand human dynamism.

Verifying the results of turing complete systems is something that Bitcoin Script can already do, so I don't think it's a particularly noteworthy feature (though I'm sure it's easier given the increased expressiveness of Simplicity).  I would personally suggest dropping the whole sentence about turing completeness, but if you want to keep it, I think s/execution/results/ would be sufficient.

File: _topics/en/simplicity.md [hunk 2]
```diff
@@ -10,14 +10,31 @@ categories:
...
+  sufficient care will need to be applied in the writing of Simplicity programs
+  to ensure they perform as intended. To use Simplicity directly on the
```
Comment (harding) [inline _topics/en/simplicity.md:19]: Maybe instead of saying that care will need to be applied to the writing of Simplicity programs, it's better to note that care still needs to be applied to the design of any contract protocol, whatever language it uses, because execution will occur based only on the code and not on intention.

File: _topics/en/simplicity.md [hunk 3]
```diff
@@ -10,14 +10,31 @@ categories:
...
+  Bitcoin blockchain requires a soft fork and any such proposal is unlikely to
+  be formalized until at least 2022-23. Currently there is Simplicity support
```
Comment (harding) [inline _topics/en/simplicity.md:21]: Nit: there's no need to use ranges with "at least"; just use the bottom of the range (e.g. "at least 2022").

File: _topics/en/simplicity.md [hunk 4]
```diff
@@ -10,14 +10,31 @@ categories:
...
+  for test branches of Bitcoin and Elements and a collection of jets are in
```
Comment (harding) [inline _topics/en/simplicity.md:22]: I don't think a collection of jets should be mention until we've described what jets are.

File: _topics/en/simplicity.md [hunk 5]
```diff
@@ -10,14 +10,31 @@ categories:
...
+  development.
 
 ## Optional.  Use Markdown formatting.  Multiple paragraphs.  Links allowed.
 extended_summary: |
-  A collection of jets, pre-made building blocks that can be combined to
-  to construct Simplicity programs, are also being developed. Jets have
-  efficient machine-code implementations for raw performance.
+  Jets are pre-made building blocks that can be combined to construct
```
Comment (harding) [inline _topics/en/simplicity.md:27]: I think this needs a few extra words to connect it to the subject mater; e.g.: "Simplicity includes support for *jets*, pre-made building blocks [...]"

File: _topics/en/simplicity.md [hunk 6]
```diff
@@ -29,21 +46,25 @@ primary_sources:
...
-    url: /en/newsletters/2020/05/06/#insert_hashtag
+    url: /en/newsletters/2020/05/06/#simplicity-next-generation-smart-contracting
     date: 2020-05-06
 
   - title: Question about implementing Taproot with Simplicity
-    url: /en/newsletters/2019/12/28/#insert_hashtag
+    url: /en/newsletters/2020/07/29/#could-we-skip-the-taproot-soft-fork-and-instead-use-simplicity-to-write-the-equivalent-of-taproot-scripts
     date: 2020-07-29
 
+  - title: BIP-Taproot
```
Comment (harding) [inline _topics/en/simplicity.md:64]: Maybe instead: "How simplicity could be implemented in a taproot leaf version"

File: _topics/en/simplicity.md [hunk 7]
```diff
@@ -29,21 +46,25 @@ primary_sources:
 ## true" to bold entry
 optech_mentions:
   - title: OP_CHECKTEMPLATEVERIFY
```
Comment (harding) [inline _topics/en/simplicity.md:48]: Simplicity is just a side note here, I don't think it's index worthy.  Suggest dropping this item.

File: _topics/en/simplicity.md [hunk 8]
```diff
@@ -29,21 +46,25 @@ primary_sources:
...
-    url: /en/newsletters/2019/12/04/#insert_hashtag
+    url: /en/newsletters/2019/12/04/#op-checktemplateverify-ctv
     date: 2019-12-04
 
   - title: Question about Simplicity and static analysis
-    url: /en/newsletters/2020/04/29/#insert_hashtag
+    url: /en/newsletters/2020/04/29/#how-is-simplicity-better-suited-for-static-analysis-compared-to-script
     date: 2020-04-29
 
   - title: Simplicity - Next Generation Smart Contracting
```
Comment (harding) [inline _topics/en/simplicity.md:56]: Suggest mentioning that this was a webinar or that it's a transcript summary.  E.g.: "Transcript: next generation smart contracting with Simplicity"

**Comment by jnewbery** [comment]
@michaelfolkson - quick reminder that there are some outstanding review comments on this and the build is failing because the intro paragraph is too long. Are you planning to work on this again?

**Comment by michaelfolkson** [comment]
@jnewbery: Yes I'll work on this Thursday and push update then (after Bitcoin Core PR review club) if that is ok.

File: _topics/en/simplicity.md (hunk 1 above)
Comment (michaelfolkson) [reply _topics/en/simplicity.md:17]: This is a really good analogy, thanks. But are there three options here for a "turing complete" program? All the nodes on the network actually execute the program to verify the execution (Ethereum), no nodes execute the program, they just verify the result meets a limited set of rules (Bitcoin Script) and a third: no one runs the program but they verify the execution was done correctly without actually doing the execution themselves (potentially Simplicity?).

I'm in two minds on whether to keep it in or not. I think it is a common question (because of Ethereum) so I'm tempted to leave it in but if you think it is better removed I'll remove.

Comment (michaelfolkson) [reply _topics/en/simplicity.md:17]: I'll remove it

**Comment by michaelfolkson** [comment]
Pushed latest changes but builds are failing (I haven't built locally yet). Will investigate.

File: _topics/en/simplicity.md (hunk 1 above)
Comment (harding) [reply _topics/en/simplicity.md]: > no nodes execute the program, they just verify the result meets a limited set of rules (Bitcoin Script) and a third: no one runs the program but they verify the execution was done correctly without actually doing the execution themselves (potentially Simplicity?)

Verification with Simplicity is the same as verification with Script, it's just that Simplicity offers a more expressive language.  It's just the difference between writing a program in one language versus another.

**Comment by harding** [comment]
@michaelfolkson 

> Is the limit for the intro paragraph 500 characters including spaces or excluding spaces? I think it is 500 characters including spaces...

Including spaces (but not the indentation).

**Comment by michaelfolkson** [comment]
Checks are passing now. It didn't like `[SIGHASH_ANYPREVOUT][topic SIGHASH_ANYPREVOUT]`. I don't know if this was because ANYPREVOUT was upper case or because ANYPREVOUT is an alias of NOINPUT and you need to use SIGHASH_NOINPUT as the topic. Anyway ready to be re-reviewed. I think I have addressed all of @harding comments.

**Comment by jnewbery** [comment]
> I don't know if this was because ANYPREVOUT was upper case or because ANYPREVOUT is an alias of NOINPUT and you need to use SIGHASH_NOINPUT as the topic.

Yes, you need to use the shortname of the topic page, or the title if there's no shortname. The title in this case is SIGHASH_NOINPUT.

**Comment by michaelfolkson** [comment]
Merged in some more @harding suggested edits. I think this is at final review stage (or at least very close to it).

**Comment by michaelfolkson** [comment]
A reminder that this is ready to be reviewed. Latest @harding edits are merged.

**Comment by harding** [comment]
ACK 931c5c1af42268b538b97242e0cabd8de6c262f3

If additional changes are requested by other people, I think maybe the simplicity.pdf link should be moved from the `see_also` section to the `primary_sources` section.  I don't think that's particularly important, though, and I feel bad about not re-reviewing this sooner as it is.
